src/ThirdPartySupport/Elementor/Traits/HasFormOptions.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Give\ThirdPartySupport\Elementor\Traits;
+
+use Exception;
+use Give\Framework\Database\DB;
+
+/**
+ * Trait to get form options with campaigns
+ *
+ * @unreleased
+ */
+trait HasFormOptions
+{
+    /**
+     * Get form options with campaigns
+     *
+     * @unreleased
+     *
+     * @return array
+     */
+    public function getFormOptionsWithCampaigns(): array
+    {
+        $campaignsWithForms = $this->getCampaignsWithForms();
+
+        if (empty($campaignsWithForms)) {
+            return [];
+        }
+
+        $campaignOptions = [];
+        $formOptionsGroup = [];
+        $campaignGroups = [];
+
+        // Group forms by campaign
+        foreach ($campaignsWithForms as $item) {
+            // Skip items without campaign association
+            if (empty($item->campaign_id) || empty($item->campaign_title)) {
+                continue;
+            }
+
+            $campaignId = $item->campaign_id;
+            $campaignTitle = $item->campaign_title;
+
+            // Add to campaign options if not already added
+            if (!isset($campaignOptions[$campaignId])) {
+                $campaignOptions[$campaignId] = $campaignTitle;
+                $campaignGroups[$campaignId] = [
+                    'label' => $campaignTitle,
+                    'options' => []
+                ];
+            }
+
+            // Add form to the campaign group
+            $campaignGroups[$campaignId]['options'][$item->id] = $item->title;
+        }
+
+        $formOptionsGroup = array_values($campaignGroups);
+
+        return $formOptionsGroup;
+    }
+
+    /**
+     * Get flattened form options from campaigns
+     *
+     * @unreleased
+     *
+     * @return array
+     */
+    public function getFormOptions(): array
+    {
+        $forms = $this->getForms();
+
+        foreach ($forms as $form) {
+            $options[$form->id] = $form->title;
+        }
+
+        return $options;
+    }
+
+    /**
+     * Query campaigns with forms
+     *
+     * @unreleased
+     *
+     * @return array
+     */
+    public function getCampaignsWithForms()
```

Comment:
Let's add the return type to this method.

Reply:
Resolved b5295e78